frontend/src/lib/components/CalendarEventsDisplay.svelte

```diff
@@ -0,0 +1,134 @@
+<svelte:options runes={true} />
+
+<script lang="ts">
+import { getEvents } from "$lib/client/sdk.gen";
+import type { CalendarEventRead } from "$lib/client/types.gen";
+import AlertMessage from "$lib/components/AlertMessage.svelte";
+import { i18n } from "$lib/i18n.svelte";
+import { Card } from "flowbite-svelte";
+import { LinkOutline } from "flowbite-svelte-icons";
+
+async function loadCalendarEvents(): Promise<
+	Record<number, CalendarEventRead[]>
+> {
+	const { data, error: apiError } = await getEvents();
+	if (apiError || data === undefined) {
+		throw new Error(
+			`Failed to load calendar events: ${apiError || "Unknown error"}`,
+		);
+	}
+
+	const sortedEvents = data.sort(
+		(a, b) =>
+			new Date(a.event_date).getTime() - new Date(b.event_date).getTime(),
+	);
+
+	const eventsByYear: Record<number, CalendarEventRead[]> = {};
+	sortedEvents.forEach((event) => {
+		const year = new Date(event.event_date).getFullYear();
+		if (!eventsByYear[year]) {
+			eventsByYear[year] = [];
+		}
+		eventsByYear[year].push(event);
+	});
+
+	return eventsByYear;
+}
+
+function formatDay(dateString: string): string {
```

Comment:
Looks like these dates get parsed as mm.dd.yyyy instead of dd.mm.yyyy?
For 19.01.2025 this page shows NaN, and 02.11 -> 11th Feb
It'd probably be more robust if we used a datetime or seconds since epoch (we do this for the milestoneanswersession already I think) in the database&API which would then get parsed without any ambiguities.